src/main/java/seedu/address/model/activity/ActivityContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.activity;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Activity} matches any of the keywords given.
+ */
+public class ActivityContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public ActivityContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        String activitiesString = person.getActivities().toString().replace("[","").replace("]", "").replace(",", "");
```

Comment:
Maybe we can find a better way to implement this, but this works for now.

Reply:
Roger, open to future fixed if anyone has a better suggestion for this method here.